Warnings in C header files are printed as errors while compiling Haskell files

When compiling the C files of the manual, a warning pops up, which is expected.

```
hs-bindgen/manual/c on  main via C v11.4.0-gcc
❯ make
gcc -DSUPPORTS_UNICODE -Wall -o manual_examples.o -c -fPIC manual_examples.c -I../../hs-bindgen/manual/c
In file included from manual_examples.c:1:
manual_examples.h:109:13: warning: `adio\U00000301s' is not in NFC [-Wnormalized=]
  109 | typedef int adiós;
      |             ^
gcc -DSUPPORTS_UNICODE -Wall -shared -o libexample.so manual_examples.o
gcc -DSUPPORTS_UNICODE -Wall -o vector.o -c -fPIC vector.c
gcc -DSUPPORTS_UNICODE -Wall -o vector_length.o -c -fPIC vector_length.c
gcc -DSUPPORTS_UNICODE -Wall -o vector_rotate.o -c -fPIC vector_rotate.c
gcc -DSUPPORTS_UNICODE -Wall -shared -o libvector.so vector.o vector_length.o vector_rotate.o -lm
gcc -DSUPPORTS_UNICODE -Wall -o game_world.o -c -fPIC game_world.c
gcc -DSUPPORTS_UNICODE -Wall -o game_player.o -c -fPIC game_player.c
gcc -DSUPPORTS_UNICODE -Wall -shared -o libgame.so game_world.o game_player.o
gcc -DSUPPORTS_UNICODE -Wall -o structs.o -c -fPIC structs.c
gcc -DSUPPORTS_UNICODE -Wall -shared -o libstructs.so structs.o
```

When we compile the Haskell files of the manual using the generated bindings for the C files we compiled before, then the same warning pops up but it's augmented with an "error" keyword and flashy red error colors (sorry, colors not visible in the output below).

```text
hs-bindgen/manual/hs on  main via λ 9.6.7
❯ cabal build all -j1
Configuration is affected by the following files:
- cabal.project
- cabal.project.local
Resolving dependencies...
Build profile: -w ghc-9.6.7 -O1
In order, the following will be built (use -v for more details):
 - c-expr-0.1.0.0 (lib:c-expr-core) (first run)
 - hs-bindgen-runtime-0.1.0 (lib) (first run)
 - c-expr-0.1.0.0 (lib) (first run)
 - hs-vector-0.1.0 (lib) (first run)
 - hs-game-0.1.0 (lib) (first run)
 - manual-0.1.0 (exe:run-manual) (first run)

...

Configuring executable 'run-manual' for manual-0.1.0...
Preprocessing executable 'run-manual' for manual-0.1.0...
Building executable 'run-manual' for manual-0.1.0...
[1 of 3] Compiling Example          ( generated/Example.hs, /home/joris/repos/haskell/hs-bindgen/manual/hs/dist-newstyle/build/x86_64-linux/ghc-9.6.7/manual-0.1.0/x/run-manual/build/run-manual/run-manual-tmp/Example.o )

In file included from /tmp/ghc241721_0/ghc_3.c:1:0: error:

/home/joris/repos/haskell/hs-bindgen/manual/c/manual_examples.h:109:13: error:
     warning: `adio\U00000301s' is not in NFC [-Wnormalized=]
      109 | typedef int adiós;
          |             ^
    |
109 | typedef int adiós;
    |             ^
[2 of 3] Compiling Structs          ( generated/Structs.hs, /home/joris/repos/haskell/hs-bindgen/manual/hs/dist-newstyle/build/x86_64-linux/ghc-9.6.7/manual-0.1.0/x/run-manual/build/run-manual/run-manual-tmp/Structs.o )
[3 of 3] Compiling RunManual        ( app/RunManual.hs, /home/joris/repos/haskell/hs-bindgen/manual/hs/dist-newstyle/build/x86_64-linux/ghc-9.6.7/manual-0.1.0/x/run-manual/build/run-manual/run-manual-tmp/RunManual.o )
[4 of 4] Linking /home/joris/repos/haskell/hs-bindgen/manual/hs/dist-newstyle/build/x86_64-linux/ghc-9.6.7/manual-0.1.0/x/run-manual/build/run-manual/run-manual
```

This can be somewhat confusing, as an error classification typically means compilation failed. In the example above, compilation completed successfully anyway. When I first tried building the manual, compilation *did* fail, but for a different reason. The warning that was classified as an error was a red herring, and the real error was:

> error while loading shared libraries: libexample.so: cannot open shared object file: No such file or director

I'll probably not be the only person that will be confused by the red herring, so we should figure out why the output looks like an error while it's actually considered a warning. My expectation is that it's something that should be fixed in GHC, but that's just a hunch.

FYI, I used `ghc-9.6.7` in the example above, but the problem is there with `ghc-9.12.2` too.

--- a/manual/hs/manual/manual.cabal
+++ b/manual/hs/manual/manual.cabal
@@ -36,3 +36,8 @@ executable run-manual
   other-modules:
       Example
       Structs
+
+  ghc-options:
+      -- Avoid gcc warnings about non-NFC characters
+      -- See <https://github.com/well-typed/hs-bindgen/issues/918>
+      -optc -Wno-normalized

